src/fork_exec/spawn.rs

```diff
@@ -114,7 +114,10 @@ impl PosixSpawner {
 
         // Set our flags.
         let mut flags: i32 = 0;
-        flags |= libc::POSIX_SPAWN_SETSIGDEF;
+        // Bug of cygwin: `posix_spawn` reports error if `POSIX_SPAWN_SETSIGDEF` is set.
+        if cfg!(not(target_os = "cygwin")) {
+            flags |= libc::POSIX_SPAWN_SETSIGDEF;
```

Comment:
I would expect this means that signals aren't correctly reset in Cygwin, which could cause some problems.

The only solution I can imagine is to not use posix_spawn on cygwin.

Reply:
This flag makes cygwin trying to set handler for SIGINT, and cause errors. I think it never worked before in the C++ version. The legacy fork + exec works fine.

Reply:
As this code stands, cygwin could potentially use posix_spawn but not reset signals, which could then lead to issues in launched commands.

Unless setsigdef is somehow not needed for cygwin, that means posix_spawn should be disabled entirely for cygwin.